rmf_traffic/src/rmf_traffic/Conflict.cpp

```diff
@@ -121,55 +121,121 @@ std::vector<ConflictData> DetectConflict::between(
 {
-  std::size_t min_size = std::min(trajectory_a.size(), trajectory_b.size());
-  if(min_size < 2)
+  Eigen::Vector2d min;
+  Eigen::Vector2d max;
+};
+
+double evaluate_spline(
+    const Eigen::Vector4d& coeffs,
+    const double t)
+{
+  // Assume time is parameterized [0,1]
+  return (coeffs[3] * t * t * t
+      + coeffs[2] * t * t
+      + coeffs[1] * t
+      + coeffs[0]);
+}
+
+std::array<double, 2> get_local_extrema(
+    const Eigen::Vector4d& coeffs)
+{
+  std::vector<double> extrema_candidates;
+  // Store boundary values as potential extrema
+  extrema_candidates.emplace_back(evaluate_spline(coeffs, 0));
+  extrema_candidates.emplace_back(evaluate_spline(coeffs, 1));
+
+  // When derivate of spline motion is not quadratic
+  if (std::abs(coeffs[3]) < 1e-12)
   {
-    throw invalid_trajectory_error::Implementation
-        ::make_segment_num_error(min_size);
+    if (std::abs(coeffs[2]) > 1e-12)
+    {
+      double t = -coeffs[1] / (2 * coeffs[2]);
+      extrema_candidates.emplace_back(evaluate_spline(coeffs, t));
+    }
   }
+  else
+  {
+    // Calculate the discriminant otherwise
+    double D = (4 * pow(coeffs[2], 2) - 12 * coeffs[3] * coeffs[1]);
 
-  if(trajectory_a.get_map_name() != trajectory_b.get_map_name())
-    return false;
 
-  const auto* t_a0 = trajectory_a.start_time();
-  const auto* t_bf = trajectory_b.finish_time();
+    if (std::abs(D) < 1e-12)
+    {
+      double t = (-2 * coeffs[2]) / (6 * coeffs[3]);
+      double extrema = evaluate_spline(coeffs, t);
+      extrema_candidates.emplace_back(extrema);
+    }
+    else if (D < 0)
+    {
+      assert(false);
+    }
+    else
+    {
+      double t1 = ((-2 * coeffs[2]) + std::sqrt(D)) / (6 * coeffs[3]);
+      double t2 = ((-2 * coeffs[2]) - std::sqrt(D)) / (6 * coeffs[3]);
 
-  // Neither of these can be null, because both trajectories should have at
-  // least two elements.
-  assert(t_a0 != nullptr);
-  assert(t_bf != nullptr);
+      extrema_candidates.emplace_back(evaluate_spline(coeffs, t1));
+      extrema_candidates.emplace_back(evaluate_spline(coeffs, t2));
+    }
+  }
+  
+  std::array<double, 2> extrema;
+  assert(!extrema_candidates.empty());
+  extrema[0] = *std::min_element(
+      extrema_candidates.begin(),
+      extrema_candidates.end());
+  extrema[1] = *std::max_element(
+      extrema_candidates.begin(),
+      extrema_candidates.end());
+
+  return extrema;
+}
 
-  if(*t_bf < *t_a0)
+BoundingBox get_bounding_box(const rmf_traffic::Spline& spline)
+{
+  BoundingBox bounding_box;
+
+  auto params = spline.get_params();
+  std::array<double, 2> extrema_x = get_local_extrema(params.coeffs[0]);
+  std::array<double, 2> extrema_y =  get_local_extrema(params.coeffs[1]);
+
+  Eigen::Vector2d min_coord = Eigen::Vector2d{extrema_x[0], extrema_y[0]};
+  Eigen::Vector2d max_coord = Eigen::Vector2d{extrema_x[1], extrema_y[1]};
+
+  double char_length =  params.profile_ptr->get_shape()
+      ->get_characteristic_length();
+
+  if (char_length > 0)
```

Comment:
How about replacing this if-statement with `assert(char_length >= 0.0);`?

If we ever come across a negative characteristic length, that's a bug. But a characteristic length of zero could be valid for a "particle" shape.

Reply:
Got it. Have update accordingly.